File: jour_21/jour21.pas
program CompterLettres;
var
  mot: string;
  nb_lettre, i: integer;
begin
  write('Entrez un mot : ');
  readln(mot);
  nb_lettre := 0;
  for i := 1 to length(mot) do
  begin
    if mot[i] <> ' ' then
      nb_lettre := nb_lettre + 1;
  end;
  writeln('Le mot ', mot, ' contient ', nb_lettre, ' lettre');
end.

File: jour_21/jour21.c
#include <stdio.h>
int main() {
    char phrase[50];
    int nb_lettre = 0;
    printf("Entrez une phrase : ");
    // fgets lit jusqu'à la fin de la ligne, espaces compris
    fgets(phrase, sizeof(phrase), stdin);
    for (int i = 0; phrase[i] != '\0'; i++) {
        // On vérifie si c'est une lettre et non un espace ou un retour à la ligne
        if (phrase[i] != ' ' && phrase[i] != '\n') {
            nb_lettre += 1;
        }
    }
    printf("La phrase contient %d lettres (hors espaces).\n", nb_lettre);
    return 0;
}